dynamic_analysis_pkg/dynamic_analysis_pkg/__init__.py

```diff
@@ -0,0 +1,167 @@
+        L.info('Fetching results of [%s]. Please wait ...', job_name)
+        storage = job.working_dir
+        self.files_list = storage.listdir()
+
+    def get_sim_results(self):
+        self.download_file_to_storage('BlueConfig')
+        self.download_report()
+        self.download_file_to_storage('out.dat')
+        L.info('Result were saved at: %s', self.local_dir)
+
+    def define_paths(self, sim_id, working_directory, new_circuit_path):
+        if not Path(new_circuit_path).is_dir():
+            raise ValueError('[ERROR] Circuit path does not exists')
+        else:
+            self.circuit_dir = new_circuit_path
+
+        self.local_dir = Path(working_directory) / Path(sim_id)
+        self.local_dir.mkdir(parents=True, exist_ok=True)
+
+        self.hpc_dir_path = Path(self.HPC_DIR_BASE) / Path(sim_id)
+
+    def download_file_to_storage(self, file_name):
+        ''' function to download the file and add it to kernel file system '''
+        file_output_path = Path(self.local_dir) / Path(file_name)
+        if file_output_path.exists():
+            L.info('- [%s] file already exists. Skipping download.', file_name)
+            return
+
+        L.info('- Fetching [%s] ...', file_name)
+        current_file = self.files_list[file_name]
+
+        if file_name == 'BlueConfig':
+            file_content = current_file.raw().read()
+            bc_str = file_content.decode('utf-8')
+            writable_content = bc_str.replace(self.ORIGIN_CIRCUIT_PATH, self.circuit_dir)
+            writable_content = writable_content.replace(str(self.hpc_dir_path), str(self.local_dir))
+
+            with open(file_output_path, 'w') as fd:
+                fd.write(writable_content)
+
+        else:
+            current_file.download(file_name) # moves to home always
+            move(file_name, file_output_path)
+
+        L.info('- [%s] downloaded', file_name)
+
+    def download_report(self):
+        reports = [x for x in self.files_list if x.endswith('.bbp')]
+        if not reports:
+            L.info('No reports were found')
+            return
+        for report in reports:
+            self.download_file_to_storage(report)
+
+
+class FetchMultipleResults:
+    '''
+    Fetch results based on a list of job_ids
+    '''
+    def __init__(self, token, sim_list_ids, base_working_directory=None):
+        self.values = []
+        for sim in tqdm(sim_list_ids):
+            self.values.append(Results(token, sim, base_working_directory))
+
+
+class AJobFinder:
+    '''
+    List the jobs launched using Uncicore in the Service Account Piz-Daint
+    '''
+    def __init__(self, token):
+        self.sorted_jobs = None
+        self.sorted_results = None
+        tr = unicore_client.Transport(token)
+        client = unicore_client.Client(tr, UNICORE_ENDPOINT)
+        self.jobs = client.get_jobs()
+
+    def show_list(self, amount_jobs_to_show=1000):
+        full_jobs_list = []
+        fetched_index = 0
```

Comment:
You still don't need this.
You can simply use `pandas.DataFrame.reset_index()` to return you a dataframe where the index has been converted as a column.